zkwrapper/source/zk_queue.cc

```diff
@@ -41,22 +42,33 @@ bool pop(const std::shared_ptr <ZKWrapper> &zk, const std::string &q_path, std::
 		LOG(ERROR) << "Failed to get children of queue node: " << q_path;
 		return false;
 	}
+
+	ZKLock q_lock(*zk, q_path);
+	q_lock.lock();
+
 	if (children.size() < 1) {
-		std::cerr << "ERROR: Queue is empty, nothing to pop!" << std::endl;
+		LOG(ERROR) << "Queue is empty, nothing to pop!";
 		// TODO: set a custom error code?
```

Comment:
I think it would make sense to use ZNONODE error, since there is no node to pop.